tesseract/tesseract_kinematics/include/tesseract_kinematics/core/validate.h

```diff
@@ -142,55 +200,111 @@ inline bool checkKinematics(const tesseract_kinematics::ForwardKinematics::Const
 {
   // Check name
   if (fwd_kin1->getName() != fwd_kin2->getName())
+  {
+    CONSOLE_BRIDGE_logError("checkKinematics: Manipulator names do not match!");
     return false;
+  }
 
   // Check if number of joints
   if (fwd_kin1->numJoints() != fwd_kin2->numJoints())
+  {
+    CONSOLE_BRIDGE_logError("checkKinematics: Manipulator number of joints do not match!");
     return false;
+  }
 
   // Check link names
-  if (fwd_kin1->getLinkNames().size() != fwd_kin2->getLinkNames().size() ||
-      !std::equal(fwd_kin1->getLinkNames().begin(), fwd_kin1->getLinkNames().end(), fwd_kin2->getLinkNames().begin()))
+  std::vector<std::string> l1 = fwd_kin1->getLinkNames();
+  std::vector<std::string> l2 = fwd_kin2->getLinkNames();
+  std::sort(l1.begin(), l1.end());
+  std::sort(l2.begin(), l2.end());
+  if (l1.size() != l1.size())
+  {
+    CONSOLE_BRIDGE_logWarn("checkKinematics: Manipulator link names size do not match!");
     return false;
+  }
+
+  if (!std::equal(l1.begin(), l1.end(), l2.begin()))
+  {
+    CONSOLE_BRIDGE_logWarn("checkKinematics: Manipulator link names do not match!");
+    return false;
```

Comment:
Perhaps this would be a useful utility?

```
bool isIdentical(const std::vector<std::string>& vec_1, const std::vector<std::string>& vec_2, std::string name = "")
```

Reply:
I like it. I am adding it to tesseract_common utils.